File: 1288. remove-covered-intervals.java
import java.util.Arrays;

class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) ->
            a[0] == b[0] ? b[1] - a[1] : a[0] - b[0]
        );

        int count = 0;
        int maxEnd = 0;

        for (int[] interval : intervals) {
            if (interval[1] > maxEnd) {
                count++;
                maxEnd = interval[1];
            }
        }

        return count;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        // Example 1
        int[][] intervals1 = {{1, 4}, {3, 6}, {2, 8}};
        System.out.println(sol.removeCoveredIntervals(intervals1)); // 2

        // Example 2
        int[][] intervals2 = {{1, 4}, {2, 3}};
        System.out.println(sol.removeCoveredIntervals(intervals2)); // 1
    }
}

File: 1288. remove-covered-intervals.py
from typing import List


class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: (x[0], -x[1]))

        count = 0
        max_end = 0

        for start, end in intervals:
            if end > max_end:
                count += 1
                max_end = end

        return count


if __name__ == "__main__":
    sol = Solution()

    # Example 1
    intervals = [[1, 4], [3, 6], [2, 8]]
    print(sol.removeCoveredIntervals(intervals))  # 2

    # Example 2
    intervals = [[1, 4], [2, 3]]
    print(sol.removeCoveredIntervals(intervals))  # 1